### PR TITLE
feat(clusters): add the cloud provider selection view

### PR DESCRIPTION
**Pull request by ctjhoa**
# What does this PR do?

https://qovery.atlassian.net/browse/FRT-1306

### CODE REVIEW

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Qovery can create a Preview Environment for this PR.
To trigger its creation, please post a comment with one of the following command.

| Command | Blueprint environment |
|---|---|
| /qovery []() preview cc1de7de-94e5-40f8-8e45-bc8986d9dfec | [storybook](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/cc1de7de-94e5-40f8-8e45-bc8986d9dfec/services/general) |
 /qovery []() preview 28c47145-c8e7-4b9d-8d9e-c65c95b48425 | [staging](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28c47145-c8e7-4b9d-8d9e-c65c95b48425/services/general) |
| /qovery []() preview {all\|UUID1,UUID2,...} | To preview multiple environments |

---

This comment has been generated from [Qovery](https://qovery.com) AI :robot:.
Below, a word from its wisdom :
> There are two hard problems in computer science: naming things, cache invalidation and off-by-1 errors

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

A preview environment was automatically created via Qovery.
Click on the link below to follow its deployment and use it.
:point_right: [\[PR\] staging - feat(clusters): add the cloud provider selection view - 2024-09-27T12:09:47Z](https://console.qovery.com/organization/3d542888-3d2c-474a-b1ad-712556db66da/project/a64f6cc1-4e34-433b-b8fd-5359716fb01b/environment/28b390b9-0497-47fb-851a-548418a55ab9/services/general)

 Another comment will be posted when deployments are finished

**Comment by RemiBonnet** [comment]
[Qovery Preview](https://www.qovery.com)
---

Your preview environment has been successfully deployed !
Click on the link below to open your service:
:point_right: [console](https://1697-console-staging.zc531a994.rustrocks.cloud)

File: libs/pages/clusters/src/lib/feature/page-new-feature/page-new-feature.tsx [hunk 1]
```diff
@@ -0,0 +1,407 @@
...
+            selectedInstallationType="demo"
+            openInstallationGuideModal={() => openInstallationGuideModal({ isDemo: true })}
+          />
+        </Section>
+        <Section className="grid grid-cols-3 gap-4">
+          <div className="col-span-3 gap-2">
+            <Heading>Or choose your hosting mode</Heading>
+            <p className="text-xs text-neutral-350">Manage your infrastructure across different hosting mode.</p>
+          </div>
+          <CardCluster
```
Comment (RemiBonnet) [inline libs/pages/clusters/src/lib/feature/page-new-feature/page-new-feature.tsx:273]: Could you use an array of object to generate list of cloud providers ?

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx [hunk 2]
```diff
@@ -69,7 +78,7 @@ export function StepGeneral(props: StepGeneralProps) {
       </div>
 
       <form onSubmit={onSubmit}>
-        <div className="mb-10 text-sm">
+        <div className="mb-10 hidden text-sm">
```
Comment (RemiBonnet) [inline libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx:81]: I think you can remove it instead of hidden

**Review by RemiBonnet — CHANGES_REQUESTED** [review]
Thanks @ctjhoa !

One point, could you use the same layout for cards with icon at the right (for non-open card)?

File: libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx (hunk 2 above)
Comment (ctjhoa) [reply libs/pages/clusters/src/lib/ui/page-clusters-create/step-general/step-general.tsx]: https://github.com/Qovery/console/pull/1697/commits/e5b8fdbb167bd9a90b8e14d4b566f9976329d4b2

File: libs/pages/clusters/src/lib/feature/page-new-feature/page-new-feature.tsx (hunk 1 above)
Comment (ctjhoa) [reply libs/pages/clusters/src/lib/feature/page-new-feature/page-new-feature.tsx]: https://github.com/Qovery/console/pull/1697/commits/499f706ba53a6daded7adecb21863fe769531dbc

**Review by RemiBonnet — APPROVED** [review]

**Pull request merged**